src/metabase/api/alert.clj

```diff
@@ -0,0 +1,182 @@
+(ns metabase.api.alert
+  "/api/alert endpoints"
+  (:require [clojure.data :as data]
+            [compojure.core :refer [DELETE GET POST PUT]]
+            [medley.core :as m]
+            [metabase
+             [email :as email]
+             [util :as u]]
+            [metabase.api
+             [common :as api]
+             [pulse :as pulse-api]]
+            [metabase.email.messages :as messages]
+            [metabase.models
+             [interface :as mi]
+             [pulse :as pulse :refer [Pulse]]]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(api/defendpoint GET "/"
+  "Fetch all alerts"
+  []
+  (for [alert (pulse/retrieve-alerts)
+        :let  [can-read?  (mi/can-read? alert)
+               can-write? (mi/can-write? alert)]
+        :when (or can-read?
+                  can-write?)]
+    (assoc alert :read_only (not can-write?))))
+
+(api/defendpoint GET "/question/:id"
+  [id]
+  (for [alert (if api/*is-superuser?*
+                (pulse/retrieve-alerts-for-card id)
+                (pulse/retrieve-user-alerts-for-card id api/*current-user-id*))
+        :let  [can-read?  (mi/can-read? alert)
```

Comment:
it doesn't look like you're using the value of `can-read?` here

Reply:
Good catch, it was a mistake and should be the same as the GET "/" endpoint. I'll refactor and have them use the same function.